### PR TITLE
[issue 114] - Adding the PostgreSql template provisioner (for legacy …

### LINKED ISSUE #114: Add a PostgreSql template based provisioner.
**Is your feature request related to a problem? Please describe.**
Add a PostgreSql template based provisioner so that existing internal tests can use Intersmash.

**Describe the solution you'd like**
A set of components (*Application, *Provisioner, *ProvisionerFactory and tests) for allowing users to provision PostgreSql via template, since this is needed by internal tests.

**Describe alternatives you've considered**
Use the PostgreSql image based provisioner that Intersmash provides already, but we'd have to adjust the existing tests, which is not good in terms of return of investment.

**Additional context**
For internal tests purposes.

### PR DESCRIPTION
**Pull request by fabiobrz**
## Description
Fix #114 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] Pull Request contains a description of the changes
 - [x] Pull Request does not include fixes for multiple issues/topics
 - [x] Code is self-descriptive and/or documented
 - [x] I have implemented unit tests to cover my changes
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/provision/openshift/PostgreSQLTemplateOpenShiftProvisioner.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import cz.xtf.core.event.helpers.EventHelper;
+import cz.xtf.core.openshift.OpenShiftWaiters;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.core.waiting.failfast.FailFastCheck;
+import io.fabric8.kubernetes.api.model.KubernetesList;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.openshift.api.model.Template;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
```
Comment (marekkopecky) [inline tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/provision/openshift/PostgreSQLTemplateOpenShiftProvisioner.java:20]: Can you please add some javadoc here?

File: tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/provision/openshift/PostgreSQLTemplateOpenShiftProvisionerFactory.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package org.jboss.intersmash.tools.provision.openshift;
+
+import org.jboss.intersmash.tools.application.Application;
+import org.jboss.intersmash.tools.application.openshift.PostgreSQLTemplateOpenShiftApplication;
+import org.jboss.intersmash.tools.provision.ProvisionerFactory;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PostgreSQLTemplateOpenShiftProvisionerFactory
```
Comment (marekkopecky) [inline tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/provision/openshift/PostgreSQLTemplateOpenShiftProvisionerFactory.java:10]: same here

File: tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/provision/openshift/PostgreSQLTemplateOpenShiftProvisioner.java (hunk 1 above)
Comment (fabiobrz) [reply tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/provision/openshift/PostgreSQLTemplateOpenShiftProvisioner.java:25]: Thanks, fixed.

**Review by marekkopecky — APPROVED** [review]
Thank you, LGTM.

**Comment by marekkopecky** [comment]
It seems that there is no option to merge this PR, I see:

> Required statuses must pass before merging
> All required [statuses](https://docs.github.com/rest/commits/statuses) and check runs on this pull request must run successfully to enable automatic merging."

Can you please double-check?

**Comment by fabiobrz** [comment]
> It seems that there is no option to merge this PR, I see:
> 
> > Required statuses must pass before merging
> > All required [statuses](https://docs.github.com/rest/commits/statuses) and check runs on this pull request must run successfully to enable automatic merging."
> 
> Can you please double-check?

Thanks @marekkopecky - I checked and will try to update the settings in the future. For now I'll merge this since you approved.

Thanks!

**Pull request merged**